src/server/survey_and_program_data.R

```diff
@@ -243,22 +243,31 @@ surveyAndProgramData <- function(input, output, state, spectrumFilesState) {
         }
     })
 
-    output$downloadSurveyTemplate <- downloadTemplate(state$survey_data_human_readable(),
-                                                        glue::glue("survey-data-{spectrumFilesState$country}.csv"))
-    output$downloadProgramTemplate <- downloadTemplate(state$program_data_human_readable(),
-                                                        glue::glue("program-data-{spectrumFilesState$country}.csv"))
+    state$surveyTemplateFileName <- shiny::reactive({
+        str(spectrumFilesState$countryAndRegionName())
+        countryAndRegionName <- spectrumFilesState$countryAndRegionName()
+        gsub(" ", "", glue::glue("survey-data-{countryAndRegionName}.csv"), fixed=TRUE)
+    })
+
+    shiny::observeEvent(spectrumFilesState$countryAndRegionName(), {
```

Comment:
Found a bug - the download handler was being initialised once with the value of "country" at the time. If the country changed, the templates would still have the old country as a filename! We need to actually watch the value of "countryAndRegionName" and recreate the download handler each time it changes.